lib/streamlit/elements/widgets/radio.py

```diff
@@ -255,6 +272,9 @@ def _radio(
             label_visibility
         )
 
+        if captions is not None:
```

Comment:
Should we check that length of captions is equal to the length of options?

I saw in frontend we render captions only if they exist (so if we will have 10 options and 7 captions, the last three elements will stay without captions), but maybe it is a good idea to check that length of lists are equal and raise StreamlitAPIException otherwise.

Reply:
I'm torn - like if I only want the first couple options to have captions might be nice to just be able to pass `captions=["First", "Second"]`... going to ask Product if they have an opinion

Reply:
Confirmed w/ product - we prefer not to enforce equivalent caption length check